components/angular-tomitribe-select/src/tomitribe-select.ts

```diff
@@ -151,6 +153,26 @@ module tomitribe_select {
         }
     }
 
+    function tribeSelectLoadOnOpen() {
+        return {
+            restrict: 'A',
+            require: 'uiSelect',
+            replace: false,
+            link: link,
+        };
+
+        function link(scope, element, attrs, uiSelectCtrl) {
+            scope.$on('uis:activate', ()=> {
+                //Check if we have the function defined  in tribe-select-fetch-on-open="function". Otherwise, check if the have the refresh attribute defined in ui-select
+                let refreshFn = attrs['tribeSelectFetchOnOpen'].indexOf('tribe-select-fetch-on-open') !== 0 ? attrs['tribeSelectFetchOnOpen'] : element.find('.ui-select-dropdown').attr('refresh');
```

Comment:
Another solution you can just force trigger search change on activation which will do everything for you. Still use refresh attribute in choices for everything. Thank you.
https://github.com/angular-ui/ui-select/blob/3f097e0a717e40e6fc931b7655afcc8a3c575f91/src/uiSelectChoicesDirective.js#L63